lib/wrappers/tokens/erc20.ts

```diff
@@ -0,0 +1,62 @@
+import {
+    Contract
+} from "ethers";
+
+import {
+    Address,
+    AddressOrAddressable,
+    SupportedProvider
+} from "../../types";
+import BaseWrapper from "../../base-wrapper";
+import { ERC20 } from "../../../typechain-types";
+import { abi as ERC20ABI } from "../../../artifacts/@openzeppelin/contracts/token/ERC20/ERC20.sol/ERC20.json"
+
+export class ERC20Wrapper extends BaseWrapper {
+    constructor(
+        address: AddressOrAddressable, 
+        provider: SupportedProvider
+    ) {
+        const contract = new Contract(address, ERC20ABI, provider)
+        super(contract, provider)
+    }
+
+    async totalSupply(): Promise<bigint> {
+        return this.call.totalSupply()
+    }
+
+    async balanceOf(account: Address): Promise<bigint> {
```

Comment:
Why all these methods are necessary if they just use `this.call` directly?

Reply:
Are you suggesting all the non-mutable methods should go inside the "call" abstraction? And just expose those that actually transact?

Reply:
Fixed in 3e64ae7